SimPEG/ObjectiveFunction.py

```diff
@@ -0,0 +1,314 @@
+        )
+
+    def __radd__(self, objfct2):
+        return self+objfct2
+
+    def __mul__(self, multiplier):
+        return ComboObjectiveFunction([self], [multiplier])
+
+    def __rmul__(self, multiplier):
+        return self * multiplier
+
+    def __div__(self, denominator):
+        return self.__mul__(1./denominator)
+
+    def __truediv__(self, denominator):
+        return self.__mul__(1./denominator)
+
+    def __rdiv__(self, denominator):
+        return self.__mul__(1./denominator)
+
+
+class ComboObjectiveFunction(BaseObjectiveFunction):
+
+    _multiplier_types = (float, None, Utils.Zero) + integer_types # Directive
+
+    def __init__(self, objfcts=[], multipliers=None, **kwargs):
+
+        if multipliers is None:
+            multipliers = len(objfcts)*[1]
+
+        self._nP = '*'
+
+        assert(len(objfcts)==len(multipliers)),(
+            "Must have the same number of Objective Functions and Multipliers "
+            "not {} and {}".format(len(objfcts),len(multipliers))
+            )
+
+        def validate_list(objfctlist, multipliers):
+            for fct, mult in zip(objfctlist, multipliers):
+                assert (
+                    isinstance(fct, BaseObjectiveFunction)
+                ) , (
+                    "Unrecognized objective function type {} in objfcts. "
+                    "All entries in objfcts must inherit from "
+                    "ObjectiveFunction".format(fct.__class__.__name__)
+                )
+
+                assert(type(mult) in self._multiplier_types), (
+                    "Objective Functions can only be multiplied by a "
+                    "float, or a properties.Float, not a {}, {}".format(
+                        type(mult), mult
+                    )
+                )
+
+                if fct.nP != '*':
+                    if self._nP != '*':
+                        assert self._nP == fct.nP, (
+                            "Objective Functions must all have the same "
+                            "nP={}, not {}".format(self.nP, [f.nP for f in objfcts])
+                        )
+                    else:
+                        self._nP = fct.nP
+
+        validate_list(objfcts, multipliers)
+
+        self.objfcts = objfcts
+        self._multipliers = multipliers
+
+        super(ComboObjectiveFunction, self).__init__(**kwargs)
+
+    def __len__(self):
+        return len(self._multipliers)
+
+    def __getitem__(self,key):
+        return self.multipliers[key],self.objfcts[key]
+
+    @property
+    def __len__(self):
+        return self.objfcts.__len__
+
+    @property
+    def multipliers(self):
+        return self._multipliers
+
+    @multipliers.setter
+    def multipliers(self, value):
+        self._multipliers = value
+
+
+    def __call__(self, m, **kwargs):
+
+        f = 0.
+        for multiplier, objfct in self:
+            if isinstance(multiplier, Utils.Zero) or multiplier == 0.: # don't evaluate the fct
+                pass
```

Comment:
This could be a continue statement:

```python
for mult, objfct in self:
    if mult == 0:
        continue
    f += mult * objfct(m, **kwargs)
return f
```

I find the pass a bit strange to read is all..